#10 Add Koin to project

mmbehroozfar opened the pull request:
Added Koin to project.

Added a sample for how we can use Koin, including a screen, domain layer and data layer.

mmbehroozfar committed:
#10 Add view model dependency to libs.versions.toml

mmbehroozfar committed:
#10 Decreased koin library version to have all dependencies

mmbehroozfar committed:
#10 Add koin dependency to all modules

mmbehroozfar committed:
#10 Implement data layer koin module

mmbehroozfar committed:
#10 Implement domain layer koin module

mmbehroozfar committed:
#10 Implement main feature screen and koin module

mmbehroozfar committed:
#10 Add koin to application

MohammadBagherNoorian commented on feature/main/ui/src/main/kotlin/com/epiclabs/thinktok/main/ui/MainScreen.kt:
```diff
@@ -0,0 +1,21 @@
+package com.epiclabs.thinktok.main.ui
```
please remove the ".remove" file in this directory

mmbehroozfar replied:
Removed!

e17073b29b9ff4a7c35264f10c6de766363030e0

MohammadBagherNoorian commented on feature/main/repository/src/main/kotlin/com/epiclabs/thinktok/main/repository/di/SomethingRepositoryModule.kt:
```diff
@@ -0,0 +1,11 @@
+import com.epiclabs.thinktok.main.domain.api.SomethingRepository
+import com.epiclabs.thinktok.main.repository.SomethingRepositoryImpl
+import org.koin.dsl.module
+
```
since these files are primarily for testing Koin and might not be needed in the long term, could you please add a TODO comment to remove these test files in the future?

viamgr replied:
An issue has been made about that. https://github.com/epic-labs-org/ThinkTokAndroid/issues/30

mmbehroozfar replied:
Added the comments to remove them in #30 

6b64fb0b7a94ab437add9664468087291bd442b9

viamgr commented on app/src/main/kotlin/com/epiclabs/thinktok/MainActivity.kt:
```diff
@@ -20,8 +21,7 @@ class MainActivity : ComponentActivity() {
         setContent {
             ThinkTokTheme {
                 Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
-                    Greeting(
```
I made another issue to remove the Greeting function. I don't think it's necessary to remove it here. https://github.com/epic-labs-org/ThinkTokAndroid/issues/29

mmbehroozfar replied:
`Greeting `function is still there (line 34), and we can remove it as part of #29. 

Here, I only replaced it with `MainScreen`

viamgr reviewed (dismissed):
What a great sample! Thanks @mmbehroozfar

mmbehroozfar committed:
#10 Remove .remove file in main feature directory

mmbehroozfar committed:
#10 Add TODO comment to remove test module in another ticket

viamgr committed:
#27 Fixed conflicts

viamgr committed:
Removed redundant proguard files and refactored build types configuration

- Removed redundant `proguard-rules.pro` files from multiple modules.
- Updated `addBuildTypes` function signature in `Common.kt`.
- Refactored `addBuildTypes` function call in `AndroidModulePlugin.kt`.

viamgr committed:
Refactor: Changed the default value of `isMinifyEnabledInRelease` to `true` in `addBuildTypes` function.
